common/djangoapps/student/models.py

```diff
@@ -884,7 +913,8 @@ def is_course_full(self, course):
         """
         is_course_full = False
         if course.max_student_enrollments_allowed is not None:
-            is_course_full = self.num_enrolled_in(course.id) >= course.max_student_enrollments_allowed
+            is_course_full = self.num_enrolled_in_exclude_admins(course.id) >= course.max_student_enrollments_allowed
```

Comment:
Since this new code will be called during enrollment in all courses post-release (CCX/non-CCX), I favor adding tests around checking enrollment limits for non-CCX courses as well. Perhaps the correct place is here?:
https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/tests/test_enrollment.py
Just simple tests like the ones below would be fine.